dspy/__init__.py

```diff
@@ -26,4 +26,55 @@
 HFModel = dsp.HFModel
 
 configure = settings.configure
-context = settings.context
\ No newline at end of file
+context = settings.context
+
+
+import json
+import os
+import openai
+
+class Models:
+    """ Singleton to easily create LMs and keep track of all LMs throughout DSPy."""
+    _instance = None
+    lms = {}
+
+    def __new__(cls, config_path=None):
+        """Singleton constructor. Optionally takes a path to a config which is uses to instantiate LMs."""
+        if not cls._instance:
+            cls._instance = super(Models, cls).__new__(cls)
+            cls._instance._configure_dspy(config_path=config_path)
+        return cls._instance
+
+    @classmethod
+    def create_lm(cls, model_config):
+        """ Create an LM based."""
+        model_config = {**model_config} # deepcopy config
+        model_name = model_config.pop('model')
+
+        if "gpt-" in model_name:
```

Comment:
create_lm could call any model provider you want. Mainly, I wanted to solve the problem of LM configuration saving and loading. You can't serialize the LM so you need to save the kwargs and easily map it to an LM. Any other idea on how to solve this?